http://gcc.gnu.org/PR27889

2006-07-20  Richard Henderson  <rth@redhat.com>

	PR 27889
	* tree-nested.c (create_tmp_var_for): Set DECL_COMPLEX_GIMPLE_REG_P
	if needed.

--- branches/gcc-4_1-branch/gcc/tree-nested.c	2006/07/20 15:07:25	115620
+++ branches/gcc-4_1-branch/gcc/tree-nested.c	2006/07/20 15:43:44	115621
@@ -143,6 +143,9 @@
   DECL_CONTEXT (tmp_var) = info->context;
   TREE_CHAIN (tmp_var) = info->new_local_var_chain;
   DECL_SEEN_IN_BIND_EXPR_P (tmp_var) = 1;
+  if (TREE_CODE (type) == COMPLEX_TYPE)
+    DECL_COMPLEX_GIMPLE_REG_P (tmp_var) = 1;
+
   info->new_local_var_chain = tmp_var;
 
   return tmp_var;
